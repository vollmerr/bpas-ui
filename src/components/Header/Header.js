import React from 'react';
import PropTypes from 'prop-types';

import {
  Grid,
} from 'react-bootstrap';

import { HeaderUtility, HeaderNav } from 'bpas-ui';

// import './style.css';

/** Renders the Utility Header */
function Header({
  title,
  links,
}) {
  return (
    <header>
      <HeaderUtility title={title} />
      <HeaderNav links={links} />
    </header>
  );
}

Header.propTypes = {
  links: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.string,
      href: PropTypes.string,
      icon: PropTypes.string,
    }),
  ),
};

export default Header;
