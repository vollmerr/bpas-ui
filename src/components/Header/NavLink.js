import React from 'react';
import PropTypes from 'prop-types';
import { NavLink as Link } from 'react-router-dom';
import styled from 'styled-components';
import Icon from '../Icon';
import theme from '../../util/theme';

const Wrapper = styled.li`
    text-align: ${props => props.isMobile ? 'left' : 'center'};
    flex: 1;

    &:not(:last-child) {
      border-bottom: ${props => props.isMobile ? `1px solid ${theme.color.primaryFaded}` : 'none'};
    }
`;

const StyledIcon = styled(Icon)`
    display: ${props => props.isMobile ? 'inline-block' : 'block'};
    vertical-align: ${props => props.isMobile ? 'middle' : 'baseline'};
    color: ${theme.color.primaryFaded};
    padding-right: ${props => props.isMobile ? '10px' : '0'};
`;

const StyledLink = styled(Link)`
    display: block;
    color: inherit;
    padding: ${theme.size.sm}px ${theme.size.xl}px;

    &:hover,
    &:focus,
    &.active,
    &:hover > ${StyledIcon},
    &:focus > ${StyledIcon},
    &.active > ${StyledIcon} {
      text-decoration: none;
      color: ${theme.color.primary};
    }
`;

/** Renders a single navbar link */
function NavLink({
  link,
  isMobile,
  onClick,
}) {
  return (
    <Wrapper isMobile={isMobile}>
      <StyledLink
        exact={link.href === '/'}
        to={link.href}
        onClick={onClick}
      >
        <StyledIcon
          icon={link.icon}
          size={32}
          isMobile={isMobile}
        />
        {link.text}
      </StyledLink>
    </Wrapper>
  );
}

NavLink.PropTypes = {
  link: PropTypes.shape({
    text: PropTypes.string.isRequired,
    href: PropTypes.string.isRequired,
    icon: PropTypes.string.isRequired,
  }).isRequired,
  isMobile: PropTypes.bool,
  onClick: PropTypes.func,
};

export default NavLink;
