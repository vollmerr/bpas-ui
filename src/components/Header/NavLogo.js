import React from 'react';
import styled from 'styled-components';
import logo from './_odiLogo.png';

const Wrapper = styled.a`
  margin: ${props => props.isMobile ? '15px 0' : '0'};
  display: inline-block;
  height: 60px;
`;

const Image = styled.img`
  height: 60px;
`;

/** Renders the navbar logo  */
function NavLogo(props) {
  return (
    <Wrapper href={'/'} {...props}>
      <Image src={logo} alt={'CA Department of Technology logo'} />
    </Wrapper>
  );
}

export default NavLogo;
