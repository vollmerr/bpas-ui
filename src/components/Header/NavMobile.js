import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import theme from '../../util/theme';
import Icon from '../Icon';


const Menu = styled.a`
  color: ${theme.color.primary};
  cursor: pointer;
  float: right;
  margin-top: 1.2em;

  &:hover {
    text-decoration: none;
  }
`;

/** Renders the navbar logo  */
function NavMobile({
  onClick,
}) {
  return (
    <Menu onClick={onClick} >
      <Icon icon={'menu'} size={36} />
    </Menu>
  );
}

NavMobile.PropTypes = {
  onClick: PropTypes.func.isRequired,
};

export default NavMobile;
