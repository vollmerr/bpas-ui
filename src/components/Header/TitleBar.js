import React from 'react';
import PropTypes from 'prop-types';
import { Grid } from 'react-bootstrap';
import styled from 'styled-components';
import theme from '../../theme';

import TitleText from './TitleText';
import TitleLinks from './TitleLinks';

const Wrapper = styled.div`
  display: flex;
  justify-content: center;
  background: ${theme.color.primary};
  height: ${theme.size.titleBar}px;
`;

const Center = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

/** Renders the headers title bar */
function TitleBar({
  title,
  isMobile = false,
}) {
  return (
    <Wrapper>
      <Grid>
        <Center>
          <TitleText title={title} isMobile={isMobile} />
          <TitleLinks />
        </Center>
      </Grid>
    </Wrapper>
  );
}

TitleBar.propTypes = {
  title: PropTypes.string.isRequired,
  isMobile: PropTypes.bool,
};

export default TitleBar;
