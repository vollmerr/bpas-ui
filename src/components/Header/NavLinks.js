import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import NavLink from './NavLink';
import theme from '../../util/theme';

const defaultLinks = [
  { text: 'Home', href: '/', icon: 'home' },
  { text: 'Help', href: '/help', icon: 'question-line' },
];

const Wrapper = styled.ul`
  display: ${props => props.isMobile ? 'block' : 'flex'};
  flex: 1;
  justify-content: space-around;
  list-style-type: none;
  margin: 0;
  padding: 0;
  z-index: 1000;
`;

/** Renders the navbar links */
function NavLinks({
  links = defaultLinks,
  isMobile = false,
  ...props,
}) {
  return (
    <Wrapper isMobile={isMobile}>
      {links.map(link => (
        <NavLink
          link={link}
          key={link.text}
          isMobile={isMobile}
          {...props}
        />
      ))}
    </Wrapper>
  );
}

NavLinks.propTypes = {
  links: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.string,
      href: PropTypes.string,
      icon: PropTypes.string,
    }),
  ),
  isMobile: PropTypes.bool,
};

export default NavLinks;
