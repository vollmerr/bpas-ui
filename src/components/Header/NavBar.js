import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import theme from '../../util/theme';

import Container from '../Container';
import NavLogo from './NavLogo';
import NavLinks from './NavLinks';
import NavMobile from './NavMobile';

const Wrapper = styled.div`
  min-width: 100%;
  background: ${theme.color.primaryInverse};
  border-bottom: 5px solid ${theme.color.second};
`;

const Content = styled.div`
  align-items: center;
  display: ${props => props.isMobile ? 'block' : 'flex'};
  justify-content: space-between;
  min-height: ${theme.size.navBar}px;
`;

/** Renders the headers title bar */
function NavBar({
  links,
  isMobile = false,
  mobileNavVisible = false,
  onClick,
  onToggle,
}) {
  const navLogoProps = {
    isMobile,
    onClick,
  };

  const navMobileProps = {
    onClick: onToggle,
  };

  const navLinksProps = {
    links,
    isMobile,
    onClick,
  };

  const displayLinks = mobileNavVisible || !isMobile;

  return (
    <Wrapper>
      <Container>
        <Content isMobile={isMobile}>
          <NavLogo {...navLogoProps} />
          {isMobile && <NavMobile {...navMobileProps} />}
          {displayLinks && <NavLinks {...navLinksProps} />}
        </Content>
      </Container>
    </Wrapper>
  );
}

NavBar.propTypes = {
  links: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.string,
      href: PropTypes.string,
      icon: PropTypes.string,
    }),
  ),
  isMobile: PropTypes.bool,
  mobileNavVisible: PropTypes.bool,
  onClick: PropTypes.func,
};

export default NavBar;
