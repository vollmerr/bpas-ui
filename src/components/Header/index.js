import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import theme from '../theme';

import TitleBar from './TitleBar';
import NavBar from './NavBar';

const Wrapper = styled.header`
  height: ${theme.size.header}px;
  margin-bottom: ${theme.size.lg}px;
  position: relative;
  z-index: ${theme.index.header}
`;

/** Renders the Header */
class Header extends Component {
  constructor(props) {
  super(props);
  this.state = {
      isMobile: false,
      mobileNavVisible: false,
    };
  }

  componentDidMount() {
    this.handleResize();
    window.addEventListener('resize', this.handleResize);
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.handleResize);
  }

  handleResize = () => {
    const mobile = window.innerWidth <= theme.breakpoints.md;

    this.setState({isMobile: mobile});

    if (!mobile) {
      this.setState({mobileNavVisible: false});
    }
  }

  handleNavToggle = () => {
    const { isMobile, mobileNavVisible } = this.state;

    if(isMobile && !mobileNavVisible) {
      this.setState({mobileNavVisible: true});
    } else {
      this.setState({mobileNavVisible: false});
    }
  }

  handleNavClose = () => {
    this.setState({mobileNavVisible: false});
  }

  render() {
    const { title } = this.props;
    const { isMobile, mobileNavVisible } = this.state;

    const titleBarProps = {
      title,
      isMobile,
      onClick: this.handleNavClose,
    };

    const navBarProps = {
      isMobile,
      mobileNavVisible,
      onClick: this.handleNavClose,
      onToggle: this.handleNavToggle,
    };

    return (
      <Wrapper>
        <TitleBar {...titleBarProps} />
        <NavBar {...navBarProps} />
      </Wrapper>
    );
  }
}

Header.propTypes = {
  title: PropTypes.string.isRequired,
  links: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.string,
      href: PropTypes.string,
      icon: PropTypes.string,
    }),
  ),
};

export default Header;
