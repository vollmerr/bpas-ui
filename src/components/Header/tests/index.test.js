import React from 'react';
import renderer from 'react-test-renderer';
import { testRouter } from '../../../utils/testUtils.js';
import Header from '../';

const testTitle = 'test title';

describe('<Header />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      testRouter(<Header title={testTitle} />)
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  xit('passes the correct props down', () => {});

  xit('toggles mobile view onToggle', () => {});

  xit('closes the mobile view onClick', () => {});
});
