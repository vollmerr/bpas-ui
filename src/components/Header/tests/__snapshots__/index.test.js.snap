// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<header
  className="sc-iRbamj fTTYAa"
>
  <div
    className="sc-hMqMXs emBvIB"
  >
    <div
      className="container"
    >
      <div
        className="sc-kEYyzF bbYXyH"
      >
        <div
          className="sc-kkGfuU fESlcU"
        >
          <a
            href="http://www.ca.gov"
            onClick={[Function]}
          >
            <img
              alt="CA Gov Logo"
              className="sc-ckVGcZ laVqmj"
              src="_caGovLogo.svg"
            />
          </a>
          <h1
            className="sc-jKJlTe jllqoO"
          >
            test title
          </h1>
        </div>
        <a
          className="sc-eNQAEJ cgLvGw"
          href="/help"
          onClick={[Function]}
        >
          Contact Us
        </a>
      </div>
    </div>
  </div>
  <div
    className="sc-jDwBTQ hUzNZn"
  >
    <div
      className="container"
    >
      <div
        className="sc-gPEVay hJupVo"
      >
        <a
          className="sc-iAyFgw kWLHyT"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="CA Department of Technology logo"
            className="sc-hSdWYo hFdoUT"
            src="_odiLogo.png"
          />
        </a>
        <ul
          className="sc-cMljjf kIGRaP"
        >
          <li
            className="sc-cvbbAY crtSCu"
          >
            <a
              className="sc-brqgnP fUWPIB"
              href="/"
              onClick={[Function]}
              style={undefined}
            >
              <span
                className="ca-gov-icon-home sc-jWBwVP iEILel sc-eHgmQL bcbtRs"
                size={32}
              />
              Home
            </a>
          </li>
          <li
            className="sc-cvbbAY crtSCu"
          >
            <a
              className="sc-brqgnP fUWPIB"
              href="/help"
              onClick={[Function]}
              style={undefined}
            >
              <span
                className="ca-gov-icon-question-line sc-jWBwVP iEILel sc-eHgmQL bcbtRs"
                size={32}
              />
              Help
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</header>
`;
